test/jdk/javax/swing/JButton/4385611/bug4385611.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2001, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Dimension;
+import java.awt.Insets;
+import java.lang.reflect.InvocationTargetException;
+import javax.swing.ImageIcon;
+import javax.swing.JButton;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+
+/*
+ * @test
+ * @bug 4385611 8078655
+ * @requires (os.family == "windows")
+ * @summary The button's preferred width/height calculation.
+ * @run main bug4385611
+ */
+
+public class bug4385611 {
+    static JButton bt1, bt2;
+    static final ImageIcon icon32x32 = new ImageIcon(bug4385611.class.getResource("red.gif"));
```

Comment:
did you verify if we are allowed to keep this red.gif in open source as per the image license? Can we not create the Imageicon from a yellow32x32 bufferedimage?

Reply:
It is just a gif file filled with yellow color created specifically for that test. It was not provided by the test submitter or anything like that - it was just created in the graphics editor. There's no copyright issue with it.

Reply:
From that point of view - sure, we can do without it. I just clarified that there is no copyright issue with the file itself.